framework/src/transfers/MultiAppReporterTransfer.C

```diff
@@ -133,15 +157,56 @@ MultiAppReporterTransfer::executeFromMultiapp()
   if (!hasFromMultiApp())
     return;
 
-  unsigned int ind = _subapp_index != std::numeric_limits<unsigned int>::max() ? _subapp_index : 0;
-  if (getFromMultiApp()->hasLocalApp(ind) &&
-      (!hasToMultiApp() || getToMultiApp()->hasLocalApp(ind)))
-    for (unsigned int n = 0; n < _from_reporter_names.size(); ++n)
-      transferReporter(_from_reporter_names[n],
-                       _to_reporter_names[n],
-                       getFromMultiApp()->appProblemBase(ind),
-                       hasToMultiApp() ? getToMultiApp()->appProblemBase(ind)
-                                       : getFromMultiApp()->problemBase());
+  std::vector<unsigned int> indices;
+  if (_distribute_reporter_vector)
+  {
+    indices.resize(getFromMultiApp()->numGlobalApps());
+    std::iota(indices.begin(), indices.end(), 0);
+  }
+  else if (_subapp_index == std::numeric_limits<unsigned int>::max())
+  {
+    indices = {0};
+  }
+  else
+    indices = {_subapp_index};
+
+  if (_distribute_reporter_vector)
+    for (unsigned int n : make_range(_to_reporter_names.size()))
+    {
+      // Clear all vector reporters and resize to the number of subapps.
+      // The summing process later will make sure the reporter values are
+      // consistent across the processors.
+      auto size = getFromMultiApp()->numGlobalApps();
+      clearVectorReporter(_to_reporter_names[n], getFromMultiApp()->problemBase());
+      resizeReporter(_to_reporter_names[n], getFromMultiApp()->problemBase(), size);
```

Comment:
so you do clear , resize, send 1 value to root, sum to synchronize

wouldnt it be cleaner to have 
send 1 value to root, synchronize all values to root value (with a new routine)

Reply:
seems this is how it s done in other places. So probably fine to keep it that way